src/views/progress_bar.rs

```diff
@@ -183,6 +188,26 @@ impl ProgressBar {
     pub fn set_value(&mut self, value: usize) {
         self.value.set(value);
     }
+
+    /// Sets the color style.
+    pub fn set_color<C>(&mut self, color: C)
+    where
+        C: Into<ColorType>,
+    {
+        self.color = color.into();
+    }
+
+    /// Sets the color style.
+    ///
+    /// Chainable variant of `set_color`.
+    pub fn with_color<C>(mut self, color: C) -> Self
+    where
+        C: Into<ColorType>,
+    {
+        self.color = color.into();
+
+        self
```

Comment:
A common trick for these methods is to use the `With` trait:

```rust
pub fn with_color<C>(mut self, color: C) -> Self
where
    C: Into<ColorType>,
{
    self.with(|s| s.set_color(color))
}
```
This way we don't need to copy code between the two methods.

Reply:
I didn't know about the `With` trait. Thanks for pointing it out!